x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -146,7 +146,20 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) {
 	}
 }
 
-func createCloudWatchEvents(getMetricDataOutput *cloudwatch.GetMetricDataOutput, instanceID string, instanceOutput ec2.Instance, regionName string) (event mb.Event, info string, err error) {
+func constructMetricQueries(listMetricsOutput []cloudwatch.Metric, instanceID string, periodInSec int) []cloudwatch.MetricDataQuery {
+	metricDataQueries := []cloudwatch.MetricDataQuery{}
+	metricDataQueryEmpty := cloudwatch.MetricDataQuery{}
+	for i, listMetric := range listMetricsOutput {
+		metricDataQuery := createMetricDataQuery(listMetric, instanceID, i, periodInSec)
+		if metricDataQuery == metricDataQueryEmpty {
+			continue
+		}
+		metricDataQueries = append(metricDataQueries, metricDataQuery)
+	}
+	return metricDataQueries
+}
```

Comment:
Also this code looks reusable between metricsets?

Reply:
Turned out this function can't be moved to a common place to reuse because it's calling createMetricDataQuery which every metricset is different hmmm

Reply:
Oh ok, maybe we could pass a function for the specific part, but let's keep it as is by now.